api/envoy/extensions/filters/http/admission_control/v3alpha/admission_control.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.admission_control.v3alpha;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/v3/range.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+import "google/rpc/status.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.admission_control.v3alpha";
+option java_outer_classname = "AdmissionControlProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Admission Control]
+// [#extension: envoy.filters.http.admission_control]
+
+message AdmissionControl {
+  // Default method of specifying what constitutes a successful request. All status codes that
+  // indicate a successful request must be explicitly specified if not relying on the default
+  // values.
+  message DefaultEvaluationCriteria {
+    // If HTTP statuses are unspecified, defaults to all non-5xx. Valid codes are in the range [100,
+    // 600).
+    repeated type.v3.Int32Range http_success_status = 1;
+
+    // GRPC status codes to consider as request successes. If unspecified, defaults to: Ok,
+    // Cancelled, Unknown, InvalidArgument, NotFound, AlreadyExists, Unauthenticated,
+    // FailedPrecondition, OutOfRange, and Unimplemented.
+    // Mappings can be found at: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md.
+    repeated uint32 grpc_success_status = 2;
+  }
+
+  // If set to false, the admission control filter will operate as a pass-through filter. If the
+  // message is unspecified, the filter will be enabled.
+  config.core.v3.RuntimeFeatureFlag enabled = 1;
+
+  // Defines how a request is considered a success/failure.
+  oneof evaluation_criteria {
+    option (validate.required) = true;
+
+    DefaultEvaluationCriteria default_eval_criteria = 2;
+  }
+
+  // The sliding time window over which the success rate is calculated. The window is rounded to the
+  // nearest second. Defaults to 120s.
+  google.protobuf.Duration sampling_window = 3;
```

Comment:
i could go either on this but if we dont actually support sub-second precision should we just use an integer instead?